### ISSUE TITLE
devDependencies (default excludeDevDependencies: 'true') not being excluded on Windows 

### ISSUE THREAD

**Issue by tmilar** [serverless/serverless#8802]
On Windows OS, the `excludeDevDependencies: true` default setting is not working, and it's probable that it has been like so for a long time. 

I've just debugged the issue a fair amount of time, since I thought it was also related to the usage of npm@7 after seeing this issue: https://github.com/serverless/serverless/issues/8500, but for my case the issue persisted even after including that version fix.
The output of the package .zip file is ~65MB on my machine, while my coworkers' using MacOS is just ~6.5MB.

Also, I've noticed about unexpected packages when inspecting the actual .zip file, ie. finding about `eslint` and `eslint-**` packages, and many more.

My debugging session concluded finding out the offending line:
https://github.com/serverless/serverless/blob/7d2ca04d7f72981c422fb8f9c25ae4699aedffba/lib/plugins/package/lib/zipService.js#L197

Notice the `\n` literal separator being used. 

The issue is that the contents of that file, were previously written in the previous exec:
https://github.com/serverless/serverless/blob/7d2ca04d7f72981c422fb8f9c25ae4699aedffba/lib/plugins/package/lib/zipService.js#L183:L185

Since the Win OS separator is `\r\n` and not `\n`, the result of the `.split('\n')` above will fail:

```
            return fs
              .readFileAsync(depFile)
              .then(fileContent => _.uniq(fileContent.toString('utf8').split(`\n`)).filter(Boolean))
              .catch(() => BbPromise.resolve());
```
This is the console.log of that `_.uniq(fileContent.toString('utf8').split(`\n`)` output. Note how all lines end with `\r`:

![image](https://user-images.githubusercontent.com/3988039/105390202-887ee780-5bf7-11eb-8b06-e6382de5bdb0.png)

<details>
<summary><b><code> sls deploy </code> output</b></summary>

```
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Installing dependencies for custom CloudFormation resources...
Serverless: WARNING: Function execute has timeout of 300 seconds, however, it's attached to API Gateway so it's automatically limited to 30 seconds.
Serverless: Deprecation warning: Safeguards support has been moved to the @serverless/safeguards-plugin external plugin and will be removed from the core with next major release.

            Please visit https://github.com/serverless/safeguards-plugin/ to migrate your safeguards to the new plugin.
            You may also disable safeguards by setting "custom.safeguards.isDisabled: true" in service config

Serverless: Safeguards Processing...
Serverless: Safeguards Results:

   Summary --------------------------------------------------

   passed - allowed-regions
   passed - allowed-runtimes
   passed - allowed-stages
   passed - framework-version
   passed - no-secret-env-vars
   passed - no-unsafe-wildcard-iam-permissions
   warned - require-cfn-role

   Details --------------------------------------------------

   1) Warned - no cfnRole set
      details: http://slss.io/sg-require-cfn-role
      Require the cfnRole option, which specifies a particular role for CloudFormation to assume while deploying.


Serverless: Safeguards Summary: 6 passed, 1 warnings, 0 errors
Serverless: Creating Stack...
Serverless: Checking Stack create progress...
........
Serverless: Stack create finished...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Uploading service pl************s.zip file to S3 (62.74 MB)...
Serverless: Uploading custom CloudFormation resources...
Serverless: Validating template...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
.............................................................................................
Serverless: Stack update finished...

...
```

</details>

<b>Installed version</b>
OS: **Windows 10** 

$ npx serverless --version
Framework Core: 1.83.2 (local)
Plugin: 3.8.4
SDK: 2.3.2
Components: 2.34.9

Since that's not the latest version, I've also tested with the latest, and the issue still persists - the offending line I've referenced above is on the current master branch.

$ npx serverless --version
Framework Core: 2.20.0 (local)
Plugin: 4.4.2
SDK: 2.3.2
Components: 3.5.1

<details>
<summary><b><code> sls deploy </code> output (on latest version)</b></summary>

```
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Installing dependencies for custom CloudFormation resources...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Uploading service p****s.zip file to S3 (68.35 MB)...
Serverless: Uploading custom CloudFormation resources...
Serverless: Validating template...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
.................
```
</details>

File size weight didn't change much: **68.35 MB**.

**Comment by pgrzesik** [comment]
I've already merged the PR but wanted to say thank you once more for providing an amazing bug report - well done sir :100:

**Comment by tmilar** [comment]
Thank you for such a great product!

**Comment by noah-potter** [comment]
I'm running into issues with this update. From what I can tell, Command Prompt will return LF line endings, whereas PowerShell will return CRLF, and you can tell the `child_process` module to use either one depending on your `ComSpec` environment variable (https://nodejs.org/api/child_process.html#child_process_default_windows_shell). My system uses Command Prompt by default, so I'm getting LF line endings which makes this update break on my machine.

It's looking like the logic should try to detect what the line endings are and split accordingly?

**Comment by pgrzesik** [comment]
Hello @noah-potter, I'm sad to hear that. Would you be able to share some docs around different line endings on Command Prompt vs Powershell? I'm surprised as I've never heard of using LF as line ending on Windows

Also, what is the current behavior for you @noah-potter ? Is it crashing?

**Comment by noah-potter** [comment]
I'm not sure what's happening and I can't find any docs that would explain what I'm seeing. The other large difference I'm seeing is that Command Prompt is using UTF-8 encoding vs PowerShell using UTF-16 LE. But again, I'm not sure why or how that might be playing into this.

It's not crashing, it's just including the devDependencies.

**Comment by pgrzesik** [comment]
Hello @noah-potter, it should be fixed in newly released `v2.21.0`. Sorry for inconvenience!

**Comment by noah-potter** [comment]
Thank you!
